server/keyshare/keyshareserver/conf.go

```diff
@@ -63,6 +63,14 @@ type Configuration struct {
 	registrationEmailTemplates map[string]*template.Template
 
 	VerificationURL map[string]string `json:"verification_url" mapstructure:"verification_url"`
+	// number of tries allowed on pin before we start with exponential backoff
+	MaxPinTries int `json:"max_pin_tries" mapstructure:"max_pin_tries"`
+	// amount of time user's email validation token is valid (in hours)
+	EmailTokenValidity int `json:"email_token_validity" mapstructure:"email_token_validity"`
+	// max number of active tokens per email address within the emailTokenRateLimitDuration
+	EmailTokenRateLimit int `json:"email_token_rate_limit" mapstructure:"email_token_rate_limit"`
+	// amount of time after which tokens become irrelevant for rate limiting (in minutes)
+	EmailTokenRateLimitDuration int `json:"email_token_rate_limit_duration" mapstructure:"email_token_rate_limit_duration"`
```

Comment:
Probably you also want to add these options to the `irma` CLI tool. Otherwise, you won't be able to actually modify these configuration options. It's also nice to specify some default values there. If no value is explicitly set (using a command line parameter or in the `--config` file), then there is a fallback.
https://github.com/privacybydesign/irmago/blob/master/irma/cmd/keyshare-server.go#L32

You probably also want to specify some test settings for these options. Otherwise the default value for `int` will be used, which is 0. 0 is not a logical value for most of these new configuration options, so that will probably cause some tests to fail.
https://github.com/privacybydesign/irmago/blob/master/internal/testkeyshare/testkeyshare.go#L60